.github/workflows/build.yml

```diff
@@ -25,14 +26,47 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install libkrb5-dev
         run: sudo apt-get -y install libkrb5-dev
-      - name: Install libxml2-dev and libxslt1-dev for python 3.9
-        # Installing lxml from source is required for python 3.9 until a precompiled Wheel archive is available from PyPI, see https://pypi.org/project/lxml/#files
-        run: sudo apt-get -y install libxml2-dev libxslt1-dev
-        if: matrix.python-version == 3.9
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
           pip install aws-adfs['test']
       - name: Test with pytest
         run: |
           pytest
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install pep517
+        run: |
+          python -m pip install pep517
+      - name: Build tagged binary wheel and source tarball
+        run: |
+          python -m pep517.build --source --binary --out-dir dist/ .
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+      - name: Build development binary wheel and source tarball
+        run: |
+          sed -i "s/^style = pep440.*/style = pep440-old/" setup.cfg # test.pypi.org does not accept local version identifiers
```

Comment:
The work-around to use `pep440-old` for development packages is not perfect as it only takes into account the number of commits since the last tag for naming the package. When branch rewrites occur, this does not work well.